clouddrift/adapters/subsurface.py

```diff
@@ -0,0 +1,170 @@
+            "expPI": (["exp"], metadata["expPI"][indices_exp]),
+            "indexExp": (["traj"], metadata["indexExp"]),
+            "fltType": (["traj"], metadata["fltType"]),
+            "id": (["traj"], metadata["indexFlt"]),
+            "rowsize": (["traj"], rowsize),
+            "time": (
+                ["obs"],
+                pd.to_datetime(data["dtnum"] - origin_datenum, unit="D"),
+            ),
+            "lon": (["obs"], data["lon"]),
+            "lat": (["obs"], data["lat"]),
+            "p": (["obs"], data["p"]),
+            "t": (["obs"], data["t"]),
+            "u": (["obs"], data["u"]),
+            "v": (["obs"], data["v"]),
+        }
+    )
+
+    # define attributes
+    vars_attrs = {
+        "expList": {
+            "long_name": "Experiment list",
+            "units": "-",
+        },
+        "expName": {
+            "long_name": "Experiment name",
+            "units": "-",
+        },
+        "expOrg": {
+            "long_name": "Experiment organization",
+            "units": "-",
+        },
+        "expPI": {
+            "long_name": "Experiment principal investigator",
+            "units": "-",
+        },
+        "indexExp": {
+            "long_name": "Experiment index number",
+            "units": "-",
+            "comment": "The index matches the float with its experiment metadata",
+        },
+        "fltType": {
+            "long_name": "Float type",
+            "units": "-",
+        },
+        "id": {"long_name": "Float ID", "units": "-"},
+        "lon": {"long_name": "Longitude", "units": "degrees_east"},
+        "lat": {"long_name": "Latitude", "units": "degrees_north"},
+        "rowsize": {
+            "long_name": "Number of observations per trajectory",
+            "sample_dimension": "obs",
+            "units": "-",
+        },
+        "u": {"long_name": "Eastward velocity", "units": "m/s"},
+        "v": {"long_name": "Northward velocity", "units": "m/s"},
+        "t": {"long_name": "Temperature", "units": "Celsius"},
+        "p": {"long_name": "Pressure", "units": "Millibar"},
+    }
+
+    # global attributes
+    attrs = {
+        "title": "Subsurface float trajectories dataset",
+        "history": SUBSURFACE_FLOAT_VERSION,
+        "date_created": datetime.now().isoformat(),
+        "publisher_name": "WOCE Subsurface Float Data Assembly Center and NOAA AOML",
+        "publisher_url": "https://www.aoml.noaa.gov/phod/float_traj/data.php",
+        "licence": "freely available",
+        "acknowledgement": f"Maintained by Andree Ramsey and Heather Furey from the Woods Hole Oceanographic Institution",
+    }
+
+    # set attributes
+    for var in vars_attrs.keys():
+        if var in ds.keys():
+            ds[var].attrs = vars_attrs[var]
+        else:
+            warnings.warn(f"Variable {var} not found in upstream data; skipping.")
+    ds.attrs = attrs
+
+    # set coordinates
+    ds = ds.set_coords(["time", "id"])
```

Comment:
Let's decide on a convention to follow. Do we want "time" and "id" or "obs" and "traj"? Do we want coordinate names to match dimension names? I think even between the datasets I implemented (MOSAiC and GLAD) I haven't been 100% consistent with this. Having a set convention would help. I'll open a separate issue for this.

Reply:
so if we follow the CF standard name, are those the variables name or just an extra attributes `"standard_name": sea_water_temperature `

Reply:
I think you meant to put this comment in the other thread about CF attributes. As I understand it, the variable name can be anything, and attributes help that variable be identified by consumers. So, having variable named `temp` with an attribute `standard_name` of `sea_water_temperature` works.